apps/web/pages/api/auth/setup/license.ts

```diff
@@ -0,0 +1,58 @@
+import type { NextApiRequest } from "next";
+import { z } from "zod";
+
+import { getServerSession } from "@calcom/features/auth/lib/getServerSession";
+import { CALCOM_PRIVATE_API_ROUTE } from "@calcom/lib/constants";
+import { HttpError } from "@calcom/lib/http-error";
+import { defaultHandler, defaultResponder } from "@calcom/lib/server";
+import prisma from "@calcom/prisma";
+
+async function checkLicenseKey(licenseKey: string) {
+  const response = await fetch(`${CALCOM_PRIVATE_API_ROUTE}/v1/license/${licenseKey}`);
+  const data = await response.json();
+  if (data.error) {
+    throw new HttpError({ statusCode: 400, message: data.error });
+  }
+
+  const licenseSchema = z.object({
+    status: z.boolean(),
+  });
+  return licenseSchema.parse(data);
+}
+
+async function saveLicenseKeyToDeployment(licenseKey: string) {
+  await prisma.deployment.upsert({
+    where: { id: 1 },
+    update: {
+      licenseKey,
+      agreedLicenseAt: new Date(),
+    },
+    create: {
+      licenseKey,
+      agreedLicenseAt: new Date(),
+    },
+  });
+}
+
+async function handler(req: NextApiRequest) {
+  const session = await getServerSession({ req });
```

Comment:
@sean-brydon after creating the admin user the session isn't set. This is why I'm getting the invalid license key error.

Reply:
Hmm very weird we do call sign i - will see what the issue is